src/modm/driver/temperature/max31855.hpp

```diff
@@ -0,0 +1,104 @@
+// coding: utf-8
+// ----------------------------------------------------------------------------
+/*
+ * Copyright (c) 2022, Sarah Vilete
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_MAX31855_HPP
+#define MODM_MAX31855_HPP
+
+#include <modm/processing/resumable.hpp>
+#include <modm/architecture/interface/spi_device.hpp>
+
+namespace modm
+{
+
+/// @ingroup modm_driver_max31855
+struct max31855
+{
+    /// Thermocouple fault bits
+    enum class
+    Fault : uint8_t
+    {
+        ShortCircuitVcc = Bit2,
+        ShortCircuitGnd = Bit1,
+        OpenCircuit     = Bit0
+    };
+
+    struct modm_packed
+    Data
+    {
+        /// @return value associated with the respective fault
+        Fault getFault()
```

Comment:
This should be callable on a `const Data` object and could also be `constexpr`.
```suggestion
        constexpr Fault getFault() const
```

Reply:
Why would this be constexpr? The return value value is not constant

Reply:
It does not have to be. It just means it can be used on a `Data` object in a `constexpr` context. C++20 even has `constexpr` support for `std::vector` with dynamic allocation at compile-time (not sure if any compiler has that implemented yet, though).
It is not that beneficial here with the intended usage of the type, but it certainly does not hurt.